lib/router.js

```diff
@@ -49,6 +49,8 @@ export default class Router {
     this.window.history[this.historyMethod] = (...args) => {
```

Comment:
we should probably prevent double proxies. i.e if something calls registerListeners twice it shouldn't create two layers of function calls.

Also would be nice that if the element is removed, it leaves window.history to the original state that it was in.

Reply:
agreed, I'll look into this

Reply:
The current router implementation relies on stacking wrapped history methods, one for each component so that each component's `this.navigate` can be called. I don't see any issue with this, and I'd rather not mess with something so fundamental.